packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -528,18 +659,153 @@ abstract class SliderComponentShape {
   /// If [labelPainter] is non-null, then [labelPainter.paint] should be
   /// called with the location that the label should appear. If the labelPainter
   /// passed is null, then no label was supplied to the [Slider].
+  void paint(
+      PaintingContext context,
+      Offset center, {
+        Animation<double> activationAnimation,
+        Animation<double> enableAnimation,
+        bool isDiscrete,
+        TextPainter labelPainter,
+        RenderBox parentBox,
+        SliderThemeData sliderTheme,
+        TextDirection textDirection,
+        double value,
+      });
+}
+
+// The following shapes are the material defaults.
+
+/// This is the default shape to a [Slider]'s track if no other shape is
+/// specified. It paints a solid colored rectangle, vertically centered in the
+/// parent box. The color is determined by the [Slider]'s enabled state and the
+/// track piece's active state. These colors are defined in the
+/// [SliderThemeData].
+///
+/// See also:
+///
+///  * [Slider] for the component that this is meant to display this shape.
+///  * [SliderThemeData] where an instance of this class is set to inform the
+///    slider of the visual details of the its track.
+///  * [SliderTrackShape] Base component for creating other custom track
+///    shapes.
+class RectangularSliderTrackShape extends SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const RectangularSliderTrackShape();
+
+  @override
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+    bool isDiscrete,
+  }) {
+    final double overlayWidth = sliderTheme.overlayShape.getPreferredSize(isEnabled, isDiscrete).width;
+    final double trackLeft = offset.dx + overlayWidth / 2;
+    final double trackTop = offset.dy + (parentBox.size.height - sliderTheme.trackHeight) / 2;
+    final double trackWidth = parentBox.size.width - overlayWidth;
+    final double trackHeight = sliderTheme.trackHeight;
+    return Rect.fromLTWH(trackLeft, trackTop, trackWidth, trackHeight);
+  }
+
+  // Spacing for disabled slider state.
+  static const double _thumbGap = 2.0;
+
+  @override
   void paint(
     PaintingContext context,
-    Offset thumbCenter, {
-    Animation<double> activationAnimation,
+    Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
     Animation<double> enableAnimation,
+    List<double> trackDividers,
+    List<bool> trackSegmentsActive,
     bool isDiscrete,
-    TextPainter labelPainter,
-    RenderBox parentBox,
+    bool isEnabled,
+  }) {
+    assert(trackDividers.isNotEmpty);
```

Comment:
I'm not sure how much it matters (if at all) but I think `assert`s normally go in the initializer list.

So `}) : assert(trackDividers.isNotEmpty)  {`

Reply:
do functions have initializer lists?

Reply:
No they do not, ignore me!

Reply:
No.  You have to do the assert the way you have it.